UndertaleModTool/Scripts/Resource Repackers/NewTextureRepacker.csx

```diff
@@ -493,24 +491,26 @@ await Task.Run(() =>
                     int potw = NearestPowerOf2((uint)item.OriginalRect.Width),
                         poth = NearestPowerOf2((uint)item.OriginalRect.Height);
 
-                    Bitmap img = new Bitmap(potw, poth, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
-
-                    // DPI fix
-                    img.SetResolution(96.0F, 96.0F);
+                    using MagickImage newAtlasImage = new(MagickColors.Transparent, potw, poth);
 
-                    Graphics g = Graphics.FromImage(img);
-
-                    // Load texture
-                    using (Bitmap source = new Bitmap(item.Filename))
-                        g.DrawImage(source, 0, 0);
+                    // Load texture, composite onto top left of new atlas
+                    using (MagickImage source = TextureWorker.ReadBGRAImageFromFile(item.Filename))
+                    {
+                        newAtlasImage.Composite(source, 0, 0, CompositeOperator.Copy);
+                    }
 
                     itemFile = Path.Combine(packagerDirectory, $"pot_{texPageItems.IndexOf(item)}.png");
-                    img.Save(itemFile, System.Drawing.Imaging.ImageFormat.Png);
+                    TextureWorker.SaveImageToFile(newAtlasImage, itemFile);
 
-                    img.Dispose();
+                    // Assign new texture image
+                    tex.TextureData.Image = GMImage.FromMagickImage(newAtlasImage).ConvertToPng(); // TODO: generate other formats
+                }
+                else
+                {
+                    // Load image from file, and assign it
+                    tex.TextureData.Image = GMImage.FromPng(File.ReadAllBytes(itemFile)); // TODO: generate other formats
```

Comment:
am confused by this, this seems to chamge the behaviour from the script?

Reply:
It's moving up the old line of code that assigned `TextureBlob`, just the true `if` branch now re-uses the existing image in memory (rather than forcing a read from disk).